src/coreclr/vm/mono/mono_coreclr.cpp

```diff
@@ -481,9 +481,33 @@ extern "C" EXPORT_API MonoClass* EXPORT_CC mono_class_from_mono_type(MonoType *i
 // Note: This function has NOT been generated by the source generator
 MonoClass * mono_class_from_name(MonoImage *image, const char* name_space, const char *name, bool ignoreCase)
 {
+    CONTRACTL
+    {
+        THROWS;
+        GC_TRIGGERS;
+        PRECONDITION(image != NULL);
+        PRECONDITION(name_space != NULL);
+        PRECONDITION(name != NULL);
+    }
+    CONTRACTL_END;
+
     GCX_PREEMP(); // temporary until we sort out our GC thread model
 
-    return g_HostStruct->class_from_name(image, name_space, name, ignoreCase);
+    MonoClass* monoClass = NULL;
+
+    EX_TRY
+    {
+        monoClass = g_HostStruct->class_from_name(image, name_space, name, ignoreCase);
+    }
+    EX_CATCH
+    {
+        SString sstr;
+        GET_EXCEPTION()->GetMessage(sstr);
+        unity_log("CoreCLR Exception %d: %s\n", GET_EXCEPTION()->GetType(), sstr.GetUTF8());
+    }
+    EX_END_CATCH(SwallowAllExceptions)
```

Comment:
If this catch was hit, it's at odds with the stance we decided with https://github.com/Unity-Technologies/runtime/pull/178  that we would make exceptions fatal.

Do we want to establish fatal vs non-fatal exceptions?

Reply:
@mrvoorhe fyi - on the 2D project the exception was not fatal. it caused load failures for objects which had not loaded types, but 80% of tests were passing. IMHO it is best to keep it fatal while we get to the stable coreclr player and editor